dbusmock/testcase.py

```diff
@@ -251,3 +278,58 @@ def spawn_server_template(cls, template: str, parameters: Dict[str, Any] = None,
                         dbus_interface=MOCK_IFACE)
 
         return (daemon, obj)
+
+    @classmethod
+    def enable_service(cls, service, system_bus: bool = False) -> bool:
+        '''Enable the given well known service name inside dbusmock
+
+        This symlinks a service file from the usual dbus service directories
+        into the dbusmock environment. Doing that allows the service to be
+        launched automatically if they are defined within $XDG_DATA_DIRS.
+
+        This function will send SIGHUP to dbus to notify it of the change.
+
+        Returns True if a service file was found and symlinked.
+        '''
+        if system_bus:
+            services_dir = 'system-services'
+            dbus_pid = cls.system_bus_pid
+        else:
+            services_dir = 'services'
+            dbus_pid = cls.session_bus_pid
+        xdg_data_dirs = os.environ.get('XDG_DATA_DIRS') or '/usr/local/share/:/usr/share/'
+
+        for d in xdg_data_dirs.split(':'):
+            src = os.path.join(d, 'dbus-1', services_dir, service + '.service')
+            if os.path.exists(src):
+                os.symlink(src, os.path.join(cls.get_services_dir(system_bus), service + '.service'))
+                # Poke dbus to reload the services
+                if dbus_pid:
+                    os.kill(dbus_pid, signal.SIGHUP)
+                return True
+
+        return False
```

Comment:
Feels more pythonic and it's also more robust to raise an exception if the service wasn't found -- that's normally unexpected in a unit test, so it shouldn't require additional code to handle that condition, but fail instead.

(You can do this very elegantly with an `else:` on the `for` loop)